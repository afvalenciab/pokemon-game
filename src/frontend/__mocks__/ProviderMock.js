import React from 'react';
import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { Router } from 'react-router';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import initialState from './initialState';
import reducer from '../reducers';

const store = createStore(reducer, initialState, compose(applyMiddleware(thunk)));
const history = createBrowserHistory();

const ProviderMock = (props) => {
  return(
    <Provider store={store}>
      <Router history={history}>
        {props.children}
      </Router>
    </Provider>
  );
};

export default ProviderMock;