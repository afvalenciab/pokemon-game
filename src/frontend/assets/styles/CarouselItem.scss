.player--carousel-item {
  display: flex;
  flex-direction: column;
  flex-shrink: 1;
  justify-content: center;
  align-items: center;
  background-color: #F2F2F2;
  width: 120px;
  height: 150px;
  margin: 0 auto;
  border-radius: 7px;
  cursor: pointer;
  transition: .3s;
  transform-origin: center left;

  figure {
    margin: 0;
    img {
      vertical-align: top;
      width: 110px;
    }
  }

  p{
    margin: 0;
    text-align: center;
  }
}

.player--carousel .player--carousel-item:hover {
  transform: scale(1.2);
  opacity: 1;
}

.player--carousel-item:hover ~ .player--carousel-item {
  transform: translateX(20px);
}