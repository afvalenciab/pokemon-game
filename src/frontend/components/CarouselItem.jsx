import React, { useState, useLayoutEffect } from 'react';
import axios from 'axios';
import PokemonCard from '../components/PokemonCard';
import logoPokemon from '../assets/static/Pokemon_logo.svg';


const CarouselItem = (props) => {
  const { name, url } = props;
  const [pokemon, setPokemon] = useState({
    data: {}
  });

  useLayoutEffect(() => {
    try {
      const fetchData = async () => {
        const response = await axios.post('/fetch', { url });
        const data = response.data;
        setPokemon({
          data
        });
      };

      fetchData();
    } catch (error) {
      
    }
  },[]);

  return Object.keys(pokemon.data).length > 0 ? (
    <>
      {pokemon.data.sprites.front_default && 
        <PokemonCard name={name} urlPicture={pokemon.data.sprites.front_default} />
      }
    </>
  ) : <PokemonCard name='Loading...' urlPicture={logoPokemon}/>;
};

export default CarouselItem;
