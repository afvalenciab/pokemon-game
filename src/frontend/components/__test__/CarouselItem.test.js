import React from 'react';
import { create } from 'react-test-renderer';
import CarouselItem from '../CarouselItem';
import ProviderMock from '../../__mocks__/ProviderMock';

describe('CarouselItem testing', () => {
  test('Match Snapshot', () => {
    const carouselItem = create(
      <ProviderMock>
        <CarouselItem />
      </ProviderMock>
    );
    expect(carouselItem.toJSON()).toMatchSnapshot();
  });
});