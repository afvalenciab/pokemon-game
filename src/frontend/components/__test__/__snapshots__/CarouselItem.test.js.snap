// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselItem testing Match Snapshot 1`] = `
<div
  className="player--carousel-item"
  onClick={[Function]}
>
  <figure>
    <img
      alt="Loading..."
      src="test-file-stubs"
    />
  </figure>
  <p>
    Loading...
  </p>
</div>
`;
